test/test-Benchmark.js

```diff
@@ -0,0 +1,129 @@
+/* globals require, exports, XPCOMUtils, Task, Services, ClientID */
+"use strict";
+
+const {before, after} = require("sdk/test/utils");
+const tabs = require("sdk/tabs");
+const simplePrefs = require("sdk/simple-prefs");
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.import("resource://gre/modules/ClientID.jsm");
+Cu.import("resource://gre/modules/Services.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+const {Benchmark} = require("./lib/Benchmark.js");
+const {getTestActivityStream} = require("./lib/utils");
+
+let app;
+let ACTIVITY_STREAMS_URL;
+
+const roundsUpTest = [
+  [0, 1],
+  [1, 1],
+  [2, 2],
+  [3, 3],
+  [5, 5],
+  [9, 10],
+  [99, 100],
+  [100, 100],
+  [140, 200],
+  [170, 200],
+  [270, 300],
+  [499, 500],
+  [500, 500],
+  [501, 1000]
+];
+
+exports.test_roundsUp = function(assert) {
+  let bench = new Benchmark(b => {});
+  for (let [n, expect] of roundsUpTest) {
+    const actual = bench._roundsUp(n);
+    assert.equal(expect, actual);
+  }
+};
+
+function fib(n) {
+  if (n < 2) {
+    return n;
+  }
+
+  return fib(n - 1) + fib(n - 2);
+}
+
+exports.test_benchmark_fib = function*(assert) {
+  let bench = new Benchmark(b => {
+    for (let i = 0; i < b.N; i++) {
+      fib(20);
+    }
+  });
+  const result = yield bench.launch();
+  assert.ok(result.ok, "it should run the benchmark and report the status");
+  assert.equal(bench._usPerOp(), result.usPerOp);
+};
+
+const openActivityStream = Task.async(function*(openUrl) {
```

Comment:
A comment here (even just pasting the above, if you want) as well as opening another issue is probably sufficient for now too.